library/core/src/main/java/com/google/android/exoplayer2/source/SimpleMediaSource.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer2.source;
+
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.Timeline;
+
+/**
+ * Single-child {@link CompositeMediaSource}.
+ */
+public abstract class SimpleMediaSource extends CompositeMediaSource<Void> {
```

Comment:
I think it might be caller to call this class `WrappingMediaSource` because "simple" doesn't really tell anyone what it's for.

Reply:
fixed